plugin/ignite-plugin/src/main/scala/org/finos/vuu/feature/ignite/schema/SchemaMapper.scala

```diff
@@ -32,33 +32,22 @@ object SchemaMapper {
   }
 
   private def hasUniqueColumnNames(columnNames: List[String]): ValidationError = {
-    if (columnNames.distinct.size != columnNames.size)
-      return Some(s"Fields map contains duplicated column names")
-
-    None
+    Option.when(columnNames.distinct.size != columnNames.size)(s"Fields map contains duplicated column names")
   }
 
   private def externalFieldsInMapConformsToExternalSchema(externalSchema: ExternalStoreEntitySchema,
                                                           externalFields: Iterable[String]): ValidationError = {
-    externalFields.foreach(field => {
-      val exists = externalSchema.schemaFields.exists(_.name == field)
-      if (!exists) return Some(s"Field `$field` not found in external schema")
-    })
-
-    None
+    externalFields
+      .find(field => externalSchema.schemaFields.forall(_.name != field))
```

Comment:
is there a reason to switch to forall rather than using exists?
Exists will exit early on first element it find that is a match rather than traversing the whole collection 
https://stackoverflow.com/questions/59922992/when-true-does-scala-exists-stop-at-the-first-element-that-satisfies-the-predic

Reply:
`forall` exits early as well, so they're same in that aspect. I just found `forall` easier to follow in this case